noko_test.rb

```diff
@@ -7,18 +7,18 @@
 # doc.xpath('//a[@class="show_full').each do |headline|
 #   puts healine.text
 
-shows = []
-
-doc.css('div.tw-plugin-full-event-list ul li div.show_full span.show_details').each do |link|
-  shows << {
-    'show_date' => doc.css('span.show_date span.tw-event-date-complete').text,
-    'show_time' => doc.css('span.show_date span.tw-event-time-complete').text}
-    puts link.content
+show_html = doc.css('div.tw-plugin-full-event-list ul li div.show_full span.show_details')
+show_details = show_html.each_with_object([]) do |link, shows|
+  shows.push({
+    date: link.css('span.show_date span.tw-event-date-complete span.tw-event-date').children.to_s.strip,
+    time: link.css('span.show_date span.tw-event-time-complete span.tw-event-time').children.to_s.strip,
+    artists: link.css('span.show_artists ul li').children.map { |el| el.to_s.strip }
```

Comment:
There might be a way to avoid the whole `#children` method to get the text out of the HTML element, I dunno that library very well, so there might be a cleaner way to do it.